lib/requestIterator.js

```diff
@@ -19,7 +20,11 @@ inherits(RequestIterator, Object)
 RequestIterator.prototype.nextRequest = function () {
   ++this.currentRequestIndex
   this.currentRequestIndex = this.currentRequestIndex < this.requests.length ? this.currentRequestIndex : 0
+  if (this.currentRequestIndex === 0) {
+    this.context = {}
+  }
```

Comment:
Can you please avoid this if and initialize this in the constructor?

Reply:
it is done as well, but the point is to clear the context when restarting the loop.

Reply:
Ah ok, can you add a comment then?

Reply:
You can include this if in the previous one. 

```js
if (this.currentRequestIndex === this.requests.length) {
  this.currentRequestIndex = 0
  // when looping over available request, clear context for a fresh start
  this.context = {}
}
```